crates/apis/src/console/mod.rs

```diff
@@ -31,50 +33,71 @@ impl JSApiSet for Console {
     }
 }
 
-fn register_console<T, U>(context: &JSContextRef, log_stream: T, error_stream: U) -> Result<()>
+fn register_console<'js, T, U>(
+    context: &Context,
+    mut log_stream: T,
+    mut error_stream: U,
+) -> Result<()>
 where
     T: Write + 'static,
     U: Write + 'static,
 {
-    let console_log_callback = context.wrap_callback(console_log_to(log_stream))?;
-    let console_error_callback = context.wrap_callback(console_log_to(error_stream))?;
-    let console_object = context.object_value()?;
-    console_object.set_property("log", console_log_callback)?;
-    console_object.set_property("error", console_error_callback)?;
-    context
-        .global_object()?
-        .set_property("console", console_object)?;
+    // TODO: Revisit the callback signatures, there's a possibility that we can
+    // actually convert from anyhow::Error to quickjs::Error.
+    context.with(|this| {
+        let globals = this.globals();
+        let console = Object::new(this.clone())?;
+
+        console.set(
+            "log",
+            Function::new(
+                this.clone(),
+                MutFn::new(move |cx, args| {
+                    let (cx, args) = hold_and_release!(cx, args);
+                    log(hold!(cx.clone(), args), &mut log_stream).map_err(|e| to_js_error(cx, e))
+                }),
+            )?,
+        )?;
+
+        console.set(
+            "error",
+            Function::new(
+                this.clone(),
+                MutFn::new(move |cx, args| {
+                    let (cx, args) = hold_and_release!(cx, args);
+                    log(hold!(cx.clone(), args), &mut error_stream).map_err(|e| to_js_error(cx, e))
+                }),
+            )?,
+        )?;
+
+        globals.set("console", console)?;
+        Ok::<_, Error>(())
+    })?;
```

Comment:
I was going to ask your review anyway!

This comment is stale by now; previously here I had a `.expect()`,  but now each callback, correctly maps `anyhow::Error` to `JSError`.  Does that address your comment?

Reply:
I think because we need access to the context, we need to map the error at that level. I was trying to think of a way to dry this up, if there was an easy way (without pulling in a crate) to create a curried function with cx, then we could just pass that if it supported FunOnce I think......

The other thing I was thinking was if we could push the to_js_error into log, but that sucks since there are multiple exit points and we would be repeating map_err's again.....

Reply:
I did thought about creating a wrapper that would allow you to automatically do the error handling (similar to currying, yeah) but I didn't invest much time on it. Another option is to create a macro that handles all this boilerplate. I'd like to land this one as is since it's fairly large already and I can follow-up on this specific piece after the fact.

Reply:
Oh, so I only left the comment because of the `TODO`.

Before creating a wrapper type I’d recommend going with `thiserror`. In general I prefer using that for libraries and only use `anyhow` for leaves (apps, servers etc). That would allow you to convert _from_ different error types to your own, and then you could just implement the `IntoJs` trait on it to make rquickjs be able to deal with it :)